### ISSUE TITLE
Error with yeoman generator-angular-fullstack

### ISSUE THREAD

**Issue by yvernet** [npm/npm#7999]
Hi,

I'm trying to use yeoman with generator-angular-fullstack, but during the "npm install" I have the following issue.

```
1921 silly addNameRange     '0.12.28-beta-f65c864',
1921 silly addNameRange     '0.12.24-dev-test',
1921 silly addNameRange     '0.12.24-dev-socketio10',
1921 silly addNameRange     '0.12.24-dev-socketio10-2',
1921 silly addNameRange     '0.12.32-beta.0',
1921 silly addNameRange     '0.12.32' ] ]
1922 verbose addNamed karma@0.12.31
1923 silly addNamed semver.valid 0.12.31
1924 silly addNamed semver.validRange 0.12.31
1925 silly cache afterAdd karma@0.12.31
1926 verbose afterAdd C:\Users\Stingray\AppData\Roaming\npm-cache\karma\0.12.31\package\package.json not in flight; writing
1927 verbose afterAdd C:\Users\Stingray\AppData\Roaming\npm-cache\karma\0.12.31\package\package.json written
1928 info attempt registry request try #2 at 08:18:22
1929 http fetch GET http://registry.npmjs.org/karma-jade-preprocessor/-/karma-jade-preprocessor-0.0.11.tgz
1930 http fetch 200 http://registry.npmjs.org/karma-jade-preprocessor/-/karma-jade-preprocessor-0.0.11.tgz
1931 silly fetchAndShaCheck shasum 5a6b653c2ceba25608907797d86922c5e10cb75e
1932 verbose addTmpTarball C:\Users\Stingray\AppData\Local\Temp\npm-3584-1d05cbe8\registry.npmjs.org\karma-jade-preprocessor\-\karma-jade-preprocessor-0.0.11.tgz not in flight; adding
1933 verbose addTmpTarball already have metadata; skipping unpack for karma-jade-preprocessor@0.0.11
1934 silly cache afterAdd karma-jade-preprocessor@0.0.11
1935 verbose afterAdd C:\Users\Stingray\AppData\Roaming\npm-cache\karma-jade-preprocessor\0.0.11\package\package.json not in flight; writing
1936 verbose afterAdd C:\Users\Stingray\AppData\Roaming\npm-cache\karma-jade-preprocessor\0.0.11\package\package.json written
1937 error git clone https://github.com/DaftMonk/grunt-build-control Cloning into bare repository 'C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\https-github-com-DaftMonk-grunt-build-control-7e4d577b'...
1937 error git clone https://github.com/DaftMonk/grunt-build-control fatal: unable to access 'https://github.com/DaftMonk/grunt-build-control/': Operation timed out after 0 milliseconds with 0 out of 0 bytes received
1938 verbose stack Error: Command failed: git -c core.longpaths=true clone --template=C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror https://github.com/DaftMonk/grunt-build-control C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\https-github-com-DaftMonk-grunt-build-control-7e4d577b
1938 verbose stack Cloning into bare repository 'C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\https-github-com-DaftMonk-grunt-build-control-7e4d577b'...
1938 verbose stack fatal: unable to access 'https://github.com/DaftMonk/grunt-build-control/': Operation timed out after 0 milliseconds with 0 out of 0 bytes received
1938 verbose stack
1938 verbose stack     at ChildProcess.exithandler (child_process.js:744:12)
1938 verbose stack     at ChildProcess.emit (events.js:110:17)
1938 verbose stack     at maybeClose (child_process.js:1008:16)
1938 verbose stack     at Socket.<anonymous> (child_process.js:1176:11)
1938 verbose stack     at Socket.emit (events.js:107:17)
1938 verbose stack     at Pipe.close (net.js:476:12)
1939 verbose cwd E:\Prog\Workspace-Intellij\test
1940 error Windows_NT 6.1.7601
1941 error argv "E:\\Prog\\NodeJS\\\\node.exe" "E:\\Prog\\NodeJS\\node_modules\\npm\\bin\\npm-cli.js" "install"
1942 error node v0.12.1
1943 error npm  v2.5.1
1944 error code 128
1945 error Command failed: git -c core.longpaths=true clone --template=C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\_templates --mirror https://github.com/DaftMonk/grunt-build-control C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\https-github-com-DaftMonk-grunt-build-control-7e4d577b
1945 error Cloning into bare repository 'C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\https-github-com-DaftMonk-grunt-build-control-7e4d577b'...
1945 error fatal: unable to access 'https://github.com/DaftMonk/grunt-build-control/': Operation timed out after 0 milliseconds with 0 out of 0 bytes received
1946 error If you need help, you may report this error at:
1946 error     <http://github.com/npm/npm/issues>
1947 verbose exit [ 1, true ]
```

Any idea please ?

**Comment by smikes** [comment]
It looks like you have a git dependency which `npm` is trying to fetch, but the connection is timing out after 0 ms (which is an unusually low timeout value...)

```
1937 error git clone https://github.com/DaftMonk/grunt-build-control Cloning into bare repository 'C:\Users\Stingray\AppData\Roaming\npm-cache\_git-remotes\https-github-com-DaftMonk-grunt-build-control-7e4d577b'...
1937 error git clone https://github.com/DaftMonk/grunt-build-control fatal: unable to access 'https://github.com/DaftMonk/grunt-build-control/': Operation timed out after 0 milliseconds with 0 out of 0 bytes received
```

If you are listing this git dependency in your own package.json, then you might need to use a different repository syntax.  One of these should work:

```
git://github.com/DaftMonk/grunt-build-control 
git+https://github.com/DaftMonk/grunt-build-control 
```

But I'm not sure what's going wrong with this installation.  Can you upload your package.json as a gist https://gist.github.com and post a link to that gist here?

Thanks!

**Comment by yvernet** [comment]
I finally managed to fix my problem by modifying my package.json. For the grunt-build-control package I had something like this
```
"grunt-build-control": "DaftMonk/grunt-build-control"
```

I replaced this by
```
"grunt-build-control": "^0.4.0"
```
and it worked !

I did not mentioned it, but I was doing all of this through a proxy using CNTLM and that seems to make things more complicated (as I tried on another computer with direct access to internet and it worked like a charm).

Thanks @smikes for making me look at the right place !

Bye.
